date-mod-2.js

```diff
@@ -0,0 +1,116 @@
+'use strict';
+var dateObj = (function (){
+  var _date = null;
+  var DAYS =
+    ['Sunday',
+    'Monday',
+    'Tuesday',
+    'Wednesday',
+    'Thursday',
+    'Friday',
+    'Saturday'];
+  var MONTHS =
+    ['January',
+    'February',
+    'March',
+    'April',
+    'May',
+    'June',
+    'July',
+    'August',
+    'September',
+    'October',
+    'November',
+    'December'];
+
+  function _initializeDate(){
+    _date = new Date();
+  }
+  
+  function setDate(date){
+    if (typeof date === 'number' || date instanceof Date){
+      _date = new Date(date);
+    }
+    else if(typeof date === 'undefined'){
+      _initializeDate();
+    }
+  }
+  
+  function getDate(formatObj){
+    var month;
+    var date;
+    var year;
+    
+    if (_date === null){
+      _initializeDate();
+    }
+    if (typeof formatObj === 'undefined' || formatObj &&
+        formatObj.format === 'milliseconds'){
+      return _date.getTime();
+    }else if (formatObj && formatObj.format === 'formatted'){
+      month = getMonthName();
+      date = _date.getDate();
+      year = _date.getFullYear();
+      
+      return month + ' ' + date + ', ' + year;
+    }
+  }
+  function getDayName(){
+    var dayIndex;
+    if (_date === null){
+      _initializeDate();
+    }
+    dayIndex = _date.getDay();
+    
+    return DAYS[dayIndex];
+  }
+
+  function getMonthName(){
+    var monthIndex;
+    if (_date === null){
+      _initializeDate();
+    }
+    monthIndex = _date.getMonth();
+    
+    return MONTHS[monthIndex];
+  }
+
+  function isToday(){
+    var date;
+    var month;
+    var year;
+    var now;
+    
+    if (_date === null){
+      _initializeDate();
+    }
+    
+    date = _date.getDate();
+    month = _date.getMonth();
+    year = _date.getYear();
+    now = new Date();
+    
+    return date === now.getDate() &&
+      month === now.getMonth() &&
+      year === now.getYear();
+  }
+  
+  function isFuture(){
+    var now;
+    if (_date === null){
+      _initializeDate();
+    }
+    
+    return _date.getTime() > now.getTime();
+  }
+
+  return {
+    setDate: setDate,
+    getDate: getDate,
+    getDayName: getDayName,
+    getMonthName: getMonthName,
+    isFuture: isFuture,
+    isToday: isToday
+  };
+});
```

Comment:
Hi James
Thanks for all of your feedback. I’ve added the function invoke to this file and pushed to repo.
Sean

From: jamdon 
Sent: Tuesday, April 26, 2016 4:21 PM
To: JSCRIPT300-Spring2016/Module_2 
Cc: seancava ; Author 
Subject: Re: [JSCRIPT300-Spring2016/Module_2] Module 2 Assignment (#13)

In date-mod-2.js:

> +    if (_date === null){
> +      _initializeDate();
> +    }
> +    
> +    return _date.getTime() > now.getTime();
> +  }
> +
> +  return {
> +    setDate: setDate,
> +    getDate: getDate,
> +    getDayName: getDayName,
> +    getMonthName: getMonthName,
> +    isFuture: isFuture,
> +    isToday: isToday
> +  };
> +});
Remember that this is an immediately executing function expression, which means there needs to be a function invocation at the end here.

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub